plugins/scaleflux/sfx-nvme.c

```diff
@@ -611,7 +611,8 @@ static void show_lat_stats_myrtle(struct sfx_lat_stats_myrtle *stats, int write)
 	for (i = 0; i < 64; i++)
 		printf("Bucket %2d: %u\n", i, stats->bucket_19[i]);
 
-	printf("\nAverage latency statistics %lld\n", stats->average);
+	printf("\nAverage latency statistics %lld\n",
+	       (unsigned long long)stats->average);
```

Comment:
Wouldn't the `__SANE_USERSPACE_TYPES__` trick not work here? https://github.com/linux-nvme/libnvme/pull/389

Reply:
No, that macro is only for powerpc - this is happening for me on amd64, if I attempt to use `PRId64` for this.

The issue seems to be that I'm getting the `__u64` defined through the int-ll64.h header, which has

```
__extension__ typedef __signed__ long long __s64;
__extension__ typedef unsigned long long __u64;
```

but the `PRI` macros are based entirely on the word size, rather than int-l64/int-ll64 configuration:

```
# if __WORDSIZE == 64
#  define __PRI64_PREFIX	"l"
#  define __PRIPTR_PREFIX	"l"
# else
#  define __PRI64_PREFIX	"ll"
#  define __PRIPTR_PREFIX
# endif
```

Reply:
(having an int-ll64 configuration here seems a bit strange, so I'll investigate a little further...)

Reply:
Yep, looks like a potential headers issue, possibly fixed by http://patchwork.ozlabs.org/project/glibc/patch/20220621090951.29911-1-metan@ucw.cz/

Reply:
Oh, that explains it. The question is how we should deal with it.

Reply:
From a brief look, those existing `PRI{u,d}64` uses are where we're printing a `uint64_t`, not a `__u64` - the latter being the kernel-defined types from `int-{l,ll}64.h`.

In some cases there's a cast from the `__u64` to a `uint64_t` first, which might be more appropriate here.

Reply:
OK, pushed a new version with `uint64_t` instead. This still has the casts, which isn't ideal, but we should be casting to the equivalent type.

Reply:
As for a longer-term fix - perhaps we should only be using the `__u{8,26,32,64}` types where they're defined in kernel interfaces and/or structured, and use the stdint `uint{8,16,32,64}_t` types everywhere else.

This would be a lot of churn though